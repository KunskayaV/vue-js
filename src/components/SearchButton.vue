<template>
  <button type="button" :class="classes" @click="onClick">{{ label }}</button>
</template>

<script lang="ts" setup>
import { computed } from 'vue'

const props = defineProps<{
  label: string
  size?: 'small' | 'medium' | 'large'
}>()

const classes = computed(() => ({
  'search-button': true,
  [`search-button--${props.size || 'medium'}`]: true
}))

</script>

<style scoped>
.search-button {
  font-size: 1rem;
  font-weight: 500;
  letter-spacing: 0.8px;
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  display: inline-block;
  line-height: 1;
  background-color: #f65261;
  text-transform: uppercase;
  color: #ffffff;
}
.search-button--small {
  font-size: 0.75rem;
  padding: 10px 16px;
}
.search-button--medium {
  font-size: 0.875rem;
  padding: 11px 24px;
}
.search-button--large {
  font-size: 1rem;
  padding: 12px 36px;
}
</style>
