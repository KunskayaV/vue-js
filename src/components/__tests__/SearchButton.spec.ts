import { describe, it, expect, vi } from 'vitest'

import { mount } from '@vue/test-utils'
import SearchButton from '../SearchButton.vue'

describe('SearchButton', () => {
  it('should match snapshot', () => {
    expect(
      mount(SearchButton, { props: { label: 'Search', size: 'small' } }).html()
    ).toMatchSnapshot()
    expect(
      mount(SearchButton, { props: { label: 'Search', size: 'medium' } }).html()
    ).toMatchSnapshot()
    expect(
      mount(SearchButton, { props: { label: 'Search', size: 'large' } }).html()
    ).toMatchSnapshot()
  })

  it('should run click handler', async () => {
    const handler = vi.fn()
    const wrapper = mount(SearchButton, {
      props: { label: 'Search', size: 'small', onClick: handler }
    })

    const button = wrapper.get('button')

    button.trigger('click')

    expect(handler).toBeCalled()
  })
})
