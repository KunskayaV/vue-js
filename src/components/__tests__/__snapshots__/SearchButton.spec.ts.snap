// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchButton > should match snapshot 1`] = `"<button data-v-d2d1fd4d=\\"\\" type=\\"button\\" class=\\"search-button search-button--small\\">Search</button>"`;

exports[`SearchButton > should match snapshot 2`] = `"<button data-v-d2d1fd4d=\\"\\" type=\\"button\\" class=\\"search-button search-button--medium\\">Search</button>"`;

exports[`SearchButton > should match snapshot 3`] = `"<button data-v-d2d1fd4d=\\"\\" type=\\"button\\" class=\\"search-button search-button--large\\">Search</button>"`;
